dev-tools/pretty/Pretty.hs

```diff
@@ -40,24 +46,37 @@ main = do
         Left err -> putStrLn $ "Error: " ++ err
         Right KoreJson{term} -> do
             case runExcept $ internalisePattern DisallowAlias CheckSubsorts Nothing internalDef term of
-                Right (trm, preds, ceils, _subst, _unsupported) -> do
-                    putStrLn "Pretty-printing pattern: "
-                    putStrLn $ renderDefault $ pretty' @'[Decoded] trm
-                    putStrLn "Bool predicates: "
-                    mapM_ (putStrLn . renderDefault . pretty' @'[Decoded]) preds
-                    putStrLn "Ceil predicates: "
-                    mapM_ (putStrLn . renderDefault . pretty' @'[Decoded]) ceils
+                Right (trm, preds, ceils, subst, unsupported) -> do
+                    Text.putStrLn . renderText . vsep $
+                        [ "Pretty-printing pattern:"
+                        , pretty' @'[Decoded, Infix] trm
+                        , renderThings "Bool predicates:" preds
+                        , renderThings "Ceil predicates:" ceils
+                        , hang 2 $ "Substitution:" <> line <> showSubst subst
+                        ]
+                    unless (null unsupported) $ do
+                        putStrLn $ "...as well as " <> show (length unsupported) <> " unsupported parts:"
+                        mapM_ print unsupported
                 Left (NoTermFound _) ->
                     case runExcept $ internalisePredicates DisallowAlias CheckSubsorts Nothing internalDef [term] of
                         Left es -> error (show es)
                         Right ts -> do
-                            putStrLn "Pretty-printing predicates: "
-                            putStrLn "Bool predicates: "
-                            mapM_ (putStrLn . renderDefault . pretty' @'[Decoded]) ts.boolPredicates
-                            putStrLn "Ceil predicates: "
-                            mapM_ (putStrLn . renderDefault . pretty' @'[Decoded]) ts.ceilPredicates
-                            putStrLn "Substitution: "
-                            mapM_ (putStrLn . renderDefault . pretty' @'[Decoded]) ts.substitution
```

Comment:
Very interestingly, this just printed the _values_ of the substitution :grimacing: instead of an actual mapping. But of course it typechecks :+1: :upside_down_face: